src/uk/gov/hmcts/contino/GithubAPI.groovy

```diff
@@ -49,16 +52,47 @@ class GithubAPI {
       requestBody: "${body}",
       consoleLogResponseBody: true,
       validResponseCodes: '200')
+
+    this.cachedLabelList = []
+  }
+
+/**
+ * Refreshes this.cachedLabelList
+*/
+  def refreshLabelCache() {
+    def project = currentProject()
+    def issueNumber = currentPullRequestNumber()
+    def response = this.steps.httpRequest(httpMode: 'GET',
+      authentication: this.steps.env.GIT_CREDENTIALS_ID,
+      acceptType: 'APPLICATION_JSON',
+      contentType: 'APPLICATION_JSON',
+      url: "${API_URL}/${project}/issues/${issueNumber}/labels",
+      consoleLogResponseBody: true,
+      validResponseCodes: '200')
+
+    def json_response = new JsonSlurper().parseText(response.content)
+    this.cachedLabelList = json_response.collect( { label -> label['name'] } )
+    return this.cachedLabelList
+  }
+
+/**
+ * Check this.cachedLabelList, if empty, call getLabels() to repopulate.
+*/
+  private getLabelsFromCache() {
+    if (this.cachedLabelList.size() == 0) {
+      return refreshLabelCache()
```

Comment:
won't we still end up calling github multiple times if there's no labels?
shouldn't we use use null and init it to empty list or have a placeholder label that shows we haven't tried loading yet?

Reply:
Yeah good point - should be resolved with the latest commit.